pytorch_lightning/metrics/classification/accuracy.py

```diff
@@ -82,24 +113,35 @@ def __init__(
         self.add_state("correct", default=torch.tensor(0), dist_reduce_fx="sum")
         self.add_state("total", default=torch.tensor(0), dist_reduce_fx="sum")
 
+        if not 0 <= threshold <= 1:
+            raise ValueError("The `threshold` should lie in the [0,1] interval.")
         self.threshold = threshold
 
+        if top_k is not None and top_k <= 0:
+            raise ValueError("The `top_k` should be an integer larger than 1.")
+        self.top_k = top_k
+
+        self.subset_accuracy = subset_accuracy
```

Comment:
```suggestion
        if not 0 <= threshold <= 1:
            raise ValueError("The `threshold` should lie in the [0,1] interval.")

        if top_k is not None and top_k <= 0:
            raise ValueError("The `top_k` should be an integer larger than 1.")

        self.threshold = threshold
        self.top_k = top_k
        self.subset_accuracy = subset_accuracy
```

Reply:
is there a check for topk > num_classes??

Reply:
Yes, this check is in the formatting function. In fact it checks topk >= num_classes, as topk = num_classes is completely useless for all purposes

Reply:
awesome :+1: